### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by bergie**
Seems like our tests were not actually catching invalid schemas. This PR rectifies that.

### CODE REVIEW

File: schemata/contentblock.yaml [hunk 1]
```diff
@@ -55,9 +55,21 @@ properties:
...
+    type: string
+    format: 'date-time'
   updated_at: # FIXME: ISOdate,null
-    type: [number, "null"]
-  anyOf: [headline, text, cta, media, list]
+    type: [string, "null"]
+    format: 'date-time'
+anyOf: [
+  '$ref': 'headline.json'
+,
```
Comment (jonnor) [inline schemata/contentblock.yaml]: Use `{ '$ref': 'headline.json' }` and you don't have to have the comma on a separate line. Or at all, I think.

Comment (bergie) [reply schemata/contentblock.yaml]: It is arguable which syntax is prettier. But anyway, the main point here is to get back to point where we're actually testing things

File: spec/schemas.coffee [hunk 2]
```diff
@@ -21,7 +20,7 @@ describe 'Schemas', ->
   schemas.forEach (schema) ->
     describe "#{schema.id} (#{schema.title})", ->
       try
-        cases = lib.getExamples schemaName
+        cases = lib.getExamples path.basename schema.id, path.extname schema.id
```
Comment (jonnor) [inline spec/schemas.coffee:26]: Why, what is the difference?

**Comment by jonnor** [comment]
Was it only invalid schemas which were not caught? And why were they not caught by checking invalid examples?

File: spec/schemas.coffee (hunk 2 above)
Comment (bergie) [reply spec/schemas.coffee:26]: `getExamples` works on names only (`contentblock`), while `schema.id` is full filename (`contentblock.json`)

**Comment by bergie** [comment]
@jonnor we were not catching them because:
- JSON meta schema validation was completely wrong (not catching invalid schemas)
- Because of this we had false positives since previously we were not checking `missed`

File: spec/schemas.coffee (hunk 2 above)
Comment (jonnor) [reply spec/schemas.coffee:26]: Yes, but but `schemaName` (which was used before) is also just the name?

Comment (bergie) [reply spec/schemas.coffee:26]: Yes, but we're now in a separate loop

**Comment by bergie** [comment]
@jonnor this iteration should be good to go. Next step will be validating `required` etc

**Comment by jonnor** [comment]
@bergie What do you mean by "validating required"?

**Comment by bergie** [comment]
@jonnor going through them and making only input-side required attributes `required` so this can be actually used for validating inbound data structures. Now we have some requireds that are only required on responses

**Comment by jonnor** [comment]
@bergie Ok. Please put them into a `responseRequired` key in that case, so we have the information?

**Comment by jonnor** [comment]
Or `requiredInResponse` or whatever

**Comment by bergie** [comment]
@jonnor can we merge? This should fix a lot of stuff already

**Comment by jonnor** [comment]
Yep

**Pull request merged**